### DISCUSSION TITLE
archspec-rs: A rust implementation

### DISCUSSION THREAD

**Discussion by baszalmstra** [archspec/archspec#160] [Show and tell]
Just wanted to let you know that there is now a working cross-platform Rust implementation of this library available:

https://github.com/prefix-dev/archspec-rs

**Comment by alalazo** [comment]
Hey, thanks for letting us know. Did you consider moving the repository to the archspec organization? I'll let @tgamblin speak for that, but I assume it wouldn't be an issue to make you / other contributors admins of the rust implementation.

**Comment by tgamblin** [comment]
It’d be great to move the rust bindings here and to bring on new folks!

**Reply by baszalmstra** [reply]
Just today I added some branding to the repository. I would be happy to move the `archspec-rs` repo under the `archspec` organization as long as we can leave the branding in the readme in place. Would that be OK?

**Reply by baszalmstra** [reply]
@tgamblin We would still like to donate the project to the archspec organization. Is this still something you would be open to?

**Reply by alalazo** [reply]
@baszalmstra That would be awesome from my point of view! What would we need to do to help? Are you still willing to be a maintainer of the rust bindings?

**Reply by tgamblin** [reply]
Still a great idea IMO!

**Reply by baszalmstra** [reply]
I just want to verify that we at prefix.dev can retain admin rights and that we can keep the branding?

**Reply by baszalmstra** [reply]
@tgamblin @alalazo Friendly ping. :)

**Reply by tgamblin** [reply]
Sorry -- I missed this.

> that we at prefix.dev can retain admin rights

I don't see a problem with admin rights on the repo -- do you @alalazo?

> and that we can keep the branding?

On the branding: `archspec` isn't in LF yet, but I think we'd like to be able to put it there, along with Spack (which is already in HPSF, part of LF). If we do that, we'd have to adhere to LF's trademark policy: https://lfprojects.org/policies/trademark-policy/. Under that I think you could still say "developed with ❤️ at prefix.dev" (though you might want something like "originally developed by prefix.dev" if more people are working on it) but saying that Archspec-rs is "a prefix.dev project" when it's part of the larger Archspec OSS project would not be ok with the LF TM policy.  Basically they want to preserve the project name as the upstream brand and not have it owned by a particular entity (see the examples at the linked policy page).  If that is ok with you all eventually, I don't really see an issue. What do you think?

If we did go to LF I think we would want to have an Archspec steering committee or some governance body -- it would make sense to put someone from prefix on it.

**Reply by alalazo** [reply]
> I don't see a problem with admin rights on the repo -- do you @alalazo?

No, not at all. I think we are fine with that.

**Reply by wolfv** [reply]
Hey @tgamblin & @alalazo - OK! I think it'd be fine to remove the prefix specific branding once (or right before) it gets accepted into LF. We can live with that.

**Reply by baszalmstra** [reply]
I tried to ransfer the repository, but github did not allow me to transfer it without the ability to create public repositories in the `archspec` organization:

<img width="1100" height="1342" alt="image" src="https://github.com/user-attachments/assets/76db180e-9329-4c7f-a496-1807e195feef" />

**Reply by alalazo** [reply]
@wolfv @baszalmstra I sent you an invitation to become members in the organization. That should give you the permissions to add a new repository.

**Reply by wolfv** [reply]
Got it & accepted. Thanks!

**Reply by baszalmstra** [reply]
[Its been moved](https://github.com/archspec/archspec-rs)!

**Reply by alalazo** [reply]
Do you have admin rights on `archspec-rs`?

**Reply by baszalmstra** [reply]
I dont think I do yet no

**Reply by alalazo** [reply]
Should be fixed

**Reply by alalazo** [reply]
Can you add "Issues" to the repository? I wanted to raise the typo `depAndabot.yml" instead of "dependabot.yml" :slightly_smiling_face:

**Reply by baszalmstra** [reply]
Works! And enabled Issues!
